src/boot/efi/boot.c

```diff
@@ -2284,9 +2281,9 @@ static VOID config_load_all_entries(
 EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *sys_table) {
         _cleanup_freepool_ EFI_LOADED_IMAGE *loaded_image = NULL;
         _cleanup_(FileHandleClosep) EFI_FILE *root_dir = NULL;
+        _cleanup_(config_free) Config config;
```

Comment:
needs to initialize with `= {}`, otherwise this might end up destroying uninitialzed memory